lib/openai/http.rb

```diff
@@ -68,17 +68,30 @@ def conn(multipart: false)
     end
 
     def uri(path:)
-      OpenAI.configuration.uri_base + OpenAI.configuration.api_version + path
+      if OpenAI.configuration.api_type == :azure
+        URI.join(OpenAI.configuration.uri_base,  path).to_s + "?api-version=#{OpenAI.configuration.api_version}"
+      else
+        URI.join(OpenAI.configuration.uri_base,  OpenAI.configuration.api_version).to_s + path
```

Comment:
I considered that, but `URI.join` doesn't seem to work with multiple arguments so it would end up as `URI.join(URI.join`, which I found a bit unecessary since the `path` in determined internally and as such already always starts with `/`.

```ruby
URI.join("https://api.openai.com", "v1", "/chat").to_s
 => "https://api.openai.com/chat" 
URI.join("https://api.openai.com", "/v1", "/chat").to_s
 => "https://api.openai.com/chat" 
```